components/calendar/src/calendar.js

```diff
@@ -0,0 +1,361 @@
+                    width: 100%;
+                }
+                .calendarTable {
+                    border: none;
+                    border-collapse: collapse;
+                    width: 100%;
+                    margin-block: ${spacers.dp4};
+                }
+                .calendarTableWrapper {
+                    padding-inline: ${spacers.dp4};
+                }
+                .calendarTable th {
+                    padding: 8px 8px;
+                    background: none;
+                    font-size: 0.85em;
+                }
+                .calendarTable td {
+                    width: ${cellSize};
+                    height: 100%;
+                    height: ${cellSize};
+                    text-align: center;
+                }
+                .calendarTable th,
+                .calendarTable td,
+                .calendarTable tr {
+                    border: 0;
+                }
+                .calendarTable td span {
+                    display: inline-flex;
+                    padding: 2px 2px;
+                    justify-content: center;
+                    border: 2px solid transparent;
+                    min-width: 16px;
+                }
+                .selectedDate {
+                    width: ${width};
+                    padding: 4px 8px;
+                }
+                .selectedDate span {
+                    justify-content: center;
+                    font-size: 0.8em;
+                }
+                .label {
+                    display: flex;
+                    padding: 4px 8px;
+                    justify-content: center;
+                    min-height: 16px;
+                }
+                button {
+                    background: none;
+                    border: 0;
+                }
+
+                button[name='day'] {
+                    overflow: hidden;
+                    width: ${cellSize};
+                    height: ${cellSize};
+                    border: 2px solid transparent;
+                    border-radius: 3px;
+                    background: none;
+                    margin: ${spacers.dp2};
+                    color: ${colors.grey900};
+                }
+
+                button[name='day']:hover {
+                    background-color: ${dayHoverBackgroundColor};
+                    text-decoration: underline;
+                }
+                button[name='day'].isSelected {
+                    background-color: ${selectedDayBackgroundColor};
+                    color: white;
+                }
+                button.isToday {
+                    text-decoration: underline;
+                }
+                button.otherMonth {
+                    color: ${colors.grey600};
+                }
+            `}</style>
+        </div>
+    )
+}
+
+Calendar.defaultProps = {
+    locale: 'en-GB',
+    numberingSystem: 'latn',
+    timeZone: Intl?.DateTimeFormat?.().resolvedOptions?.()?.timeZone || 'UTC',
```

Comment:
I'm not 100% sure that in this case providing defaults here is a good idea.... As I envisioned it, the hook should be able to resolve sensible default values for all these date/time related properties. And ultimately I'd like these values to be resolved using the following prioritisation scheme:
1. Prop-values passed to the hook/component
2. Properties available from the system/user settings resource*
3. Values coming from `Intl.DateTimeFormat().resolvedOptions()`
4. Some hardcoded defaults

[*] I'm not sure this is easy to implement right now. Quite likely we need some sort of date-internationalisation provider that holds this info first. But still good to keep this in mind, and also even when skipping this step for now, what I'm suggesting still makes sense.

So if we implement things in the hook as I suggest above, then using default props here breaks the "prioritisation scheme" I've outlined above.